x/evm/predeploys_test.go

```diff
@@ -165,8 +175,99 @@ func (suite *KeeperTestSuite) TestCreate3() {
 	suite.Require().Equal(crypto.Keccak256Hash(runCodeBytes), ethAcc.GetCodeHash())
 	suite.Require().Equal(
 		suite.App.EvmKeeper.GetCode(suite.Ctx, ethAcc.GetCodeHash()),
-		runCodeBytes,
+		[]byte(runCode),
 	)
 	// no funds are generated
-	suite.Require().Equal(beforeBalance, suite.App.EvmKeeper.GetBalance(suite.Ctx, create3))
+	suite.Require().Equal(beforeBalance, suite.App.EvmKeeper.GetBalance(suite.Ctx, destination))
+}
+
+// the predeploys are blocked from receiving funds in app/app.go.
+// however, this block should not deter them from forwarding funds
+// to the contracts they deploy.
+func (suite *KeeperTestSuite) TestBalanceForwarding() {
```

Comment:
so adding a predeploy address like create3factory contract address to blocklist would only forbid the predeploy to receive funds in bank module(or similar native module), now the test shows the predeploy could forward the balance to deploted contract, but would the blocklist prevent it from receiving funds through EVM call?